src/reader.rs

```diff
@@ -2601,6 +2755,35 @@ impl<Term: Terminal> Reader<Term> {
             where T: Into<Cow<'static, str>> {
         self.word_break = chars.into();
     }
+
+    /// Creaes a sender object that allows other threads to log messages while
+    /// a prompt is open, without causing display errors. The `Reader` handles
+    /// checking the buffer, so unless `read_line` or a similar function is
+    /// running in the Reader, the messages sent will not be printed.
+    pub fn get_log_sender(&mut self) -> LogSender {
+        if let Some(ref receiver) = self.log_channel {
+            receiver.new_sender()
+        } else {
+            let receiver = LogReceiver::new();
+            let sender = receiver.new_sender();
+            self.log_channel = Some(receiver);
+            sender
+        }
+    }
+
+    /// Returns the interval at which `Reader` checks for log messages while
+    /// waiting for user input.
+    ///
+    /// The default value is `POLL_LOG_INTERVAL_MS` milliseconds.
+    pub fn poll_log_interval(&self) -> Duration {
```

Comment:
Is there a compelling reason for this to be public? It seems like an implementation detail to me.

Reply:
It is. My thought process was that I could see someone wanting to change it from the default. (Though 10ms seems reasonable, I can see it.)

Reply:
On second thought, I suppose it can stay public. It's not that big a deal. Crates don't have strict backward compatibility requirements, anyway; it could be removed if the implementation were changed.